src/main/java/org/folio/rest/impl/UsersAPI.java

```diff
@@ -530,6 +542,20 @@ private void updateUser(User entity, Map<String, String> okapiHeaders, PostgresC
       }));
   }
 
+  private Future<Void> isUsernameUpdatedAndUnique(User entity, Map<String, String> okapiHeaders, Conn conn, User userFromStorage) {
+    return ObjectUtils.notEqual(entity.getUsername(), userFromStorage.getUsername()) ? isUsernameUnique(entity, okapiHeaders, conn) : Future.succeededFuture();
+  }
+
+  private Future<Void> isUsernameUnique(User entity, Map<String, String> okapiHeaders, Conn conn) {
```

Comment:
need to add in method name something to show that it is accross tenants

Reply:
I propose also to move this logic from api layer to user tenant service

Reply:
Moved to UserTenantService